scripts/gen_gdt.py

```diff
@@ -0,0 +1,173 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2017 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import argparse
+import sys
+import struct
+import os
+from elftools.elf.elffile import ELFFile
+from elftools.elf.sections import SymbolTableSection
+
+def debug(text):
+    if not args.verbose:
+        return
+    sys.stdout.write(os.path.basename(sys.argv[0]) + ": " + text + "\n")
+
+def error(text):
+    sys.stderr.write(os.path.basename(sys.argv[0]) + ": " + text + "\n")
+    sys.exit(1)
+
+
+gdt_pd_fmt = "<HIH"
+# 6 byte pseudo descriptor, but we're going to actually use this as the
+# zero descriptor and return 8 bytes
+def create_gdt_pseudo_desc(addr, size):
+    # ...and take back one byte for the Intel god whose Ark this is...
+    size = size - 1
+    return struct.pack(gdt_pd_fmt, size, addr, 0)
+
+
+# Limit argument always in bytes
+def chop_base_limit(base, limit):
+    base_lo = base & 0xFFFF
+    base_mid = (base >> 16) & 0xFF
+    base_hi = (base >> 24) & 0xFF
+
+    limit_lo = limit & 0xFFFF
+    limit_hi = (limit >> 16) & 0xF
+
+    return (base_lo, base_mid, base_hi, limit_lo, limit_hi)
+
+
+gdt_ent_fmt = "<HHBBBB"
+
+def create_code_data_entry(base, limit, gran, ex, dpl, rw, dc):
+    present = 1;
+
+    base_lo, base_mid, base_hi, limit_lo, limit_hi = chop_base_limit(base,
+            limit)
+
+    # Cast all these to 1 or 0
+    if rw:
```

Comment:
Instead of many boolean values, consider a bitmask. It's more readable at the call site (reducing possibility of swapping values by mistake):

    create_code_data_entry(base, limit, FLAGS_EX | FLAGS_RW)

Otherwise, these statements are a no-op, as only `0` or `1` are passed from `main()`.